src/include/execution/vm/bytecode_handlers.h

```diff
@@ -1536,6 +1536,70 @@ VM_OP_WARM void OpPow(noisepage::execution::sql::Real *result, const noisepage::
   noisepage::execution::sql::ArithmeticFunctions::Pow(result, *base, *val);
 }
 
+// ---------------------------------------------------------
+// Atomic memory operations
+// ---------------------------------------------------------
+
+// NOLINTNEXTLINE (clang tidy incorrectly assesses dest as pointing to const)
+VM_OP_HOT void OpAtomicAnd1(uint8_t *dest, uint8_t val) {
+  reinterpret_cast<std::atomic<uint8_t> *>(dest)->fetch_and(val);
```

Comment:
You can't assume that `std::atomic<T>` and `T` have the same size or layout. You can, instead, use one of the GCC/Clang atomic builtins which use raw types. See [here](https://github.com/cmu-db/noisepage/blob/master/src/include/execution/sql/chaining_hash_table.h#L199) for an example. You'll need to figure out the parameters to pass to achieve the memory semantics you intended with the original `fetch_and()` call.

Reply:
I can make that change, but for at least the 8- and 64-bit cases we have static asserts that this is the case (see `concurrent_bitmap.h`).

Reply:
Please do, I'd rather not have that restriction.